packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx

```diff
@@ -166,321 +158,403 @@ export type MenuItemStylesProps = Required<
+        if (forceTriggerFocus || shouldStopPropagation) {
+          focusAsync(itemRef.current);
+        }
+      });
 
-    const shouldStopPropagation = inSubmenu || props.vertical;
-    trySetMenuOpen(false, e, () => {
       if (forceTriggerFocus || shouldStopPropagation) {
-        focusAsync(itemRef.current);
+        e.stopPropagation();
       }
-    });
+    };
 
-    if (forceTriggerFocus || shouldStopPropagation) {
-      e.stopPropagation();
-    }
-  };
-
-  const openMenu = (e: React.MouseEvent | React.KeyboardEvent) => {
-    if (menu && !menuOpen) {
-      trySetMenuOpen(true, e);
-      _.invoke(props, 'onActiveChanged', e, { ...props, active: true });
-      e.stopPropagation();
-      e.preventDefault();
-    }
-  };
-
-  const trySetMenuOpen = (
-    newValue: boolean,
-    e: MouseEvent | React.FocusEvent | React.KeyboardEvent | React.MouseEvent,
-    onStateChanged?: any,
-  ) => {
-    setMenuOpen(newValue);
-    // The reason why post-effect is not passed as callback to trySetState method
-    // is that in 'controlled' mode the post-effect is applied before final re-rendering
-    // which cause a broken behavior: for e.g. when it is needed to focus submenu trigger on ESC.
-    // TODO: all DOM post-effects should be applied at componentDidMount & componentDidUpdated stages.
-    onStateChanged && onStateChanged();
-    _.invoke(props, 'onMenuOpenChange', e, {
-      ...props,
-      menuOpen: newValue,
-    });
-  };
-
-  const menuItemInner = (
-    <Ref innerRef={itemRef}>
-      <ElementType
-        {...getA11yProps('root', {
-          className: classes.root,
-          disabled,
-          onBlur: handleBlur,
-          onFocus: handleFocus,
-          ...unhandledProps,
-        })}
-        {...(!wrapper && { onClick: handleClick })}
+    const openMenu = (e: React.MouseEvent | React.KeyboardEvent) => {
+      if (menu && !menuOpen) {
+        trySetMenuOpen(true, e);
+        _.invoke(props, 'onActiveChanged', e, { ...props, active: true });
+        e.stopPropagation();
+        e.preventDefault();
+      }
+    };
+
+    const trySetMenuOpen = (
+      newValue: boolean,
+      e: MouseEvent | React.FocusEvent | React.KeyboardEvent | React.MouseEvent,
+      onStateChanged?: any,
+    ) => {
+      setMenuOpen(newValue);
+      // The reason why post-effect is not passed as callback to trySetState method
+      // is that in 'controlled' mode the post-effect is applied before final re-rendering
+      // which cause a broken behavior: for e.g. when it is needed to focus submenu trigger on ESC.
+      // TODO: all DOM post-effects should be applied at componentDidMount & componentDidUpdated stages.
+      onStateChanged && onStateChanged();
+      _.invoke(props, 'onMenuOpenChange', e, {
+        ...props,
+        menuOpen: newValue,
+      });
+    };
+
+    const menuItemInner = (
+      <Ref
+        innerRef={node => {
+          // @ts-ignore
+          itemRef.current = node;
```

Comment:
```suggestion
          itemRef.current = node as HTMLDivElement;
```

Reply:
Casting is correct 👍 but I still need the ts-ignore because of 
```
TS2540: Cannot assign to 'current' because it is a read-only property.
```

Or am I missing something?